### DISCUSSION TITLE
Update async database with task_postrun

### DISCUSSION THREAD

**Discussion by njdowdy** [celery/celery#7500] [Q&A] [answered]
I am new to Celery and using it in a FastAPI app with an async SQLAlchemy engine (connected to postgres).

I am requesting some advice on how to update the status of a task in my db once the Celery task is done (so, using `task_postrun`). The issue I have is that I cannot call async functions in my `task_postrun`.

The workaround "solution" I came up with is to have some db operations that look up tasks with pending/started status and use the `AsyncResult` from `celery.result` to check their status and update it if completed. 

However, this approach ***asks*** each task in process if it is done, but I want each task in process to ***tell*** me it is done at the time it finishes (which is much more efficient and ensures task status is up-to-date in the db). 

Any help on implementation of passing a task uuid from `task_postrun` to an async function that can update a db would be much appreciated (or alternative ways of accomplishing the same thing!). 

Thank you!

**Comment by njdowdy** [comment, marked as answer]
After much headache, I've solved this by making a request to a FastAPI endpoint that will perform the selective status update when the task is complete with a constructed url (e.g., `app_ip:app_port/update/{task_uuid}`) inside the `task_postrun` function. 

It feels like a lame approach, so I'm interested to hear how others handle this.

**Reply by preethamnagesh8** [reply]
Have you identified a better way of doing this ? I'm stuck in a similar situation and am looking for a more elegant way to implement this

**Reply by njdowdy** [reply]
No, unfortunately not. I've been working on other projects since this time. This approach worked for my particular use case though.
